socorro/processor/breakpad_transform_rules.py

```diff
@@ -459,3 +491,153 @@ def _predicate(
+        doc='a path where temporary files may be written',
+        default=tempfile.gettempdir(),
+    )
+
+
+    #--------------------------------------------------------------------------
+    #def __init__(self, config):
+        #super(BreakpadStackwalkerRule, self).__init__(config)
+
+
+    #--------------------------------------------------------------------------
+    def version(self):
+        return '1.0'
+
+    #--------------------------------------------------------------------------
+    @contextmanager
+    def _temp_raw_crash_json_file(self, raw_crash, crash_id):
+        file_pathname = os.path.join(
+            self.config.temporary_file_system_storage_path,
+            "%s.%s.TEMPORARY.json" % (
+                crash_id,
+                threading.currentThread().getName()
+            )
+        )
+        with open(file_pathname, "w") as f:
+            ujson.dump(dict(raw_crash), f)
+        try:
+            yield file_pathname
+        finally:
+            os.unlink(file_pathname)
+
+    #--------------------------------------------------------------------------
+    def _execute_external_process(self, command_line, processor_meta):
+        stackwalker_output, return_code = super(
+            BreakpadStackwalkerRule2015,
+            self
+        )._execute_external_process(command_line, processor_meta)
+
+        if not isinstance(stackwalker_output, Mapping):
+            processor_notes.append(
+                "MDSW produced unexpected output: %s..." %
+                str(stackwalker_output)[:10]
+            )
+            stackwalker_output = {}
+
+        stackwalker_data = DotDict()
+        stackwalker_data.json_dump = stackwalker_output
+        stackwalker_data.mdsw_return_code = return_code
+
+        stackwalker_data.mdsw_status_string = stackwalker_output.get(
+            'status',
+            'unknown error'
+        )
+        stackwalker_data.success = stackwalker_data.mdsw_status_string == 'OK'
+
+        if return_code == 124:
+            processor_meta.processor_notes.append(
+                "MDSW terminated with SIGKILL due to timeout"
+            )
+        elif return_code != 0 or not stackwalker_data.success:
+            processor_meta.processor_notes.append(
+                "MDSW failed on '%s': %s" % (
+                    command_line,
+                    stackwalker_data.mdsw_status_string
+                )
+            )
+
+        return stackwalker_data, return_code
+
+    #--------------------------------------------------------------------------
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        if 'additional_minidumps' not in processed_crash:
+            processed_crash.additional_minidumps = []
+        with self._temp_raw_crash_json_file(
+            raw_crash,
+            raw_crash.uuid
+        ) as raw_crash_pathname:
+            for dump_name, dump_pathname in raw_dumps.iteritems():
```

Comment:
We might not need the `dump_pathname` so no need to use `iteritems()`. Instead we should use `iter(raw_dumps)` or `raw_dumps.iterkeys()`.